INPTPZ1/NewtonsFractalSolver.cs

```diff
@@ -0,0 +1,175 @@
+
+                double[] inputParameters = new double[4];
+                for (int i = 0; i < inputParameters.Length; i++)
+                {
+                    inputParameters[i] = double.Parse(args[i + 2]);
+                }
+
+                fileName = args[6];
+
+                image = new Bitmap(imageWidth, imageHeight);
+                xMin = inputParameters[0];
+                xMax = inputParameters[1];
+                yMin = inputParameters[2];
+                yMax = inputParameters[3];
+
+                xStep = (xMax - xMin) / imageWidth;
+                yStep = (yMax - yMin) / imageHeight;
+
+                roots = new List<ComplexNumber>();
+
+                polynomial = new Polynomial();
+                polynomial.Coefficients.Add(new ComplexNumber() { Real = 1 });
+                polynomial.Coefficients.Add(ComplexNumber.Zero);
+                polynomial.Coefficients.Add(ComplexNumber.Zero);
+                polynomial.Coefficients.Add(new ComplexNumber() { Real = 1 });
+                derivation = polynomial.Derive();
+            }
+            else
+            {
+                throw new ArgumentException("There must be at least " + minNumberOfArguments + " arguments!");
+            }
+        }
+
+        private ComplexNumber FindPixelCoordinates(int i, int j)
+        {
+            double x = xMin + j * xStep;
+            double y = yMin + i * yStep;
+
+            return new ComplexNumber()
+            {
+                Real = Math.Max(x, minValue),
+                Imaginary = Math.Max(y, minValue)
+            };
+        }
+
+        private int FindSolutionOfEquation(ref ComplexNumber coordinates)
+        {
+            int numberOfIteration = 0;
+            for (int i = 0; i < maxIterations; i++)
+            {
+                var complexNumber = polynomial.Evaluate(coordinates).Divide(derivation.Evaluate(coordinates));
+                coordinates = coordinates.Subtract(complexNumber);
+
+                if (Math.Pow(complexNumber.Real, 2) + Math.Pow(complexNumber.Imaginary, 2) >= precision)
+                {
+                    i--;
+                }
+
+                numberOfIteration++;
+            }
+
+            return numberOfIteration;
+        }
+
+        private int FindRootNumber(ComplexNumber coordinates)
+        {
+            bool known = false;
+            int numberOfRoots = 0;
+            for (int i = 0; i < roots.Count; i++)
+            {
+                if (Math.Pow(coordinates.Real - roots[i].Real, 2) + Math.Pow(coordinates.Imaginary - roots[i].Imaginary, 2) <= tolerance)
+                {
+                    known = true;
+                    numberOfRoots = i;
+                }
+            }
+
+            if (!known)
+            {
+                roots.Add(coordinates);
+                numberOfRoots = roots.Count;
+            }
+
+            return numberOfRoots;
+        }
+
+        private readonly Color[] colors = new Color[]
```

Comment:
mixing methods and attributes is poor organization of code, move attributes to the top of class

Reply:
attribute moved to another attributes